diff --git a/lib/fastifySession.js b/lib/fastifySession.js
index 37aea626f3d76db7325d0269817af5a0276e0f55..578c65067caaf19e8619fd8b5d3050b7fbcb8644 100644
--- a/lib/fastifySession.js
+++ b/lib/fastifySession.js
@@ -18,11 +18,16 @@ function session (fastify, options, next) {
   fastify.decorate('decryptSession', (sessionId, request, callback) => {
     decryptSession(sessionId, options, request, callback)
   })
-  fastify.decorateRequest('sessionStore', options.store)
-  fastify.decorateRequest('session', {})
+  fastify.decorateRequest('sessionStore', null)
+  fastify.decorateRequest('session', null)
   fastify.decorateRequest('destroySession', destroySession)
   fastify.addHook('preValidation', preValidation(options))
   fastify.addHook('onSend', onSend(options))
+  fastify.addHook('onRequest', (req, reply, next) => {
+    req.sessionStore = options.store
+    req.session = {}
+    next()
+  })
   next()
 }
 
